webapp/channels/src/components/advanced_text_editor/use_submit.tsx

```diff
@@ -0,0 +1,315 @@
+import {getChannel, getAllChannelStats} from 'mattermost-redux/selectors/entities/channels';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentUserId, getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
+
+import {scrollPostListToBottom} from 'actions/views/channel';
+import {onSubmit} from 'actions/views/create_comment';
+import {openModal} from 'actions/views/modals';
+
+import EditChannelHeaderModal from 'components/edit_channel_header_modal';
+import EditChannelPurposeModal from 'components/edit_channel_purpose_modal';
+import NotifyConfirmModal from 'components/notify_confirm_modal';
+import PostDeletedModal from 'components/post_deleted_modal';
+import ResetStatusModal from 'components/reset_status_modal';
+
+import Constants, {ModalIdentifiers, UserStatuses} from 'utils/constants';
+import {isErrorInvalidSlashCommand, isServerError, specialMentionsInText} from 'utils/post_utils';
+
+import type {GlobalState} from 'types/store';
+import type {PostDraft} from 'types/store/draft';
+
+import useGroups from './use_groups';
+
+function getStatusFromSlashCommand(message: string) {
+    const tokens = message.split(' ');
+    const command = tokens[0] || '';
+    if (command[0] !== '/') {
+        return '';
+    }
+    const status = command.substring(1);
+    if (status === 'online' || status === 'away' || status === 'dnd' || status === 'offline') {
+        return status;
+    }
+
+    return '';
+}
+
+const useSubmit = (
+    draft: PostDraft,
+    postError: React.ReactNode,
+    channelId: string,
+    postId: string,
+    serverError: (ServerError & { submittedMessage?: string }) | null,
+    lastBlurAt: React.MutableRefObject<number>,
+    focusTextbox: (forceFocust?: boolean) => void,
+    setServerError: (err: (ServerError & { submittedMessage?: string }) | null) => void,
+    setPostError: (err: React.ReactNode) => void,
+    setShowPreview: (showPreview: boolean) => void,
+    handleDraftChange: (draft: PostDraft, options?: {instant?: boolean; show?: boolean}) => void,
+    prioritySubmitCheck: (onConfirm: () => void) => boolean,
+): [
+        (e: React.FormEvent, submittingDraft?: PostDraft) => void,
+        string | null,
+    ] => {
+    const getGroupMentions = useGroups(channelId, draft.message);
+
+    const dispatch = useDispatch();
+
+    const isDraftSubmitting = useRef(false);
+    const [errorClass, setErrorClass] = useState<string | null>(null);
+    const isDirectOrGroup = useSelector((state: GlobalState) => {
+        const channel = getChannel(state, channelId);
+        return channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL;
+    });
+
+    const channel = useSelector((state: GlobalState) => {
+        return getChannel(state, channelId);
+    }); // TODO: TRY TO AVOID THIS
```

Comment:
Is this TODO left over from the old code or is it still needed?

Reply:
I didn't like to depend on the whole channel object, since it has many things and may cause uneeded re-renders. There is another todo in this file related to that.

But maybe it is an early optimization. We can remove the todos and see how it behaves.

Reply:
Yeah, I'd prefer to remove the TODO for now. This sort of improvement is always good to keep in mind, but it's easy for us to forget about them until years later when no one remembers what we wanted to change or why